src/jit/lowerxarch.cpp

```diff
@@ -2377,6 +2377,14 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
         return;
     }
 
+    // TODO-XArch-CQ: improve table driven number of arguments encoding
+    // Due to different number of arguments for different base type for the same
+    // hardware intrinsic function we need to special case NI_SSE2_Shuffle
```

Comment:
We can update function `Compiler::numArgOfHWIntrinsic`.

Reply:
Yes. It will be necessary to support other intrinsics with variable number of arguments like `SetVector128` / `SetVector256`. Perhaps we could use flag like `HW_Flag_VariableNumberOfArgs` to short circuit numArg selection

Reply:
> variable number of arguments like SetVector128 / SetVector256

Not really, these two guys are "NoCodgen" herplers, we can use `sig->numArg` in the importer.